toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -57,10 +60,179 @@ func handleKernelCommandLine(extraCommandLine string, imageChroot *safechroot.Ch
+
+	case imagecustomizerapi.SELinuxPermissive, imagecustomizerapi.SELinuxEnforcing:
+		newSELinuxArgs = installutils.CmdlineSELinuxSettings
+
+	case imagecustomizerapi.SELinuxDisabled:
+		newSELinuxArgs = ""
+
+	default:
+		return fmt.Errorf("unknown SELinux mode (%s)", selinuxMode)
+	}
+
+	// Insert/replace/delete SELinux command-line arguments.
+	newGrub2Config := grub2Config[:start] + newSELinuxArgs + grub2Config[end:]
+
+	// Update grub.cfg file.
+	err = writeGrub2ConfigFile(newGrub2Config, imageChroot)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getCurrentSELinuxMode(imageChroot *safechroot.Chroot) (imagecustomizerapi.SELinux, error) {
+	logger.Log.Debugf("Get existing SELinux mode")
+
+	grub2Config, err := readGrub2ConfigFile(imageChroot)
+	if err != nil {
+		return "", err
+	}
+
+	// Try to find any existing SELinux args.
+	selinuxArgsMatch := linuxCommandLineSELinuxRegex.FindStringSubmatchIndex(grub2Config)
+	if selinuxArgsMatch == nil {
+		// SELinux not present in the kernel command-line.
+		// Therefore, it is disabled.
+		return imagecustomizerapi.SELinuxDisabled, nil
+	}
+
+	// Check if "enforcing=1" is present.
+	// Note: A negative index means the sub-match is not present.
+	// Note: regexp returns index pairs. So, [4] is start index of the 2nd sub-match.
+	if selinuxArgsMatch[4] >= 0 {
+		return imagecustomizerapi.SELinuxForceEnforcing, nil
+	}
+
+	selinuxMode, err := getSELinuxModeFromConfigFile(imageChroot)
+	if err != nil {
+		return "", err
+	}
+
+	return selinuxMode, nil
+}
+
+func getSELinuxModeFromConfigFile(imageChroot *safechroot.Chroot) (imagecustomizerapi.SELinux, error) {
+	selinuxConfigFilePath := filepath.Join(imageChroot.RootDir(), installutils.SELinuxConfigFile)
+
+	// Read the SELinux config file.
+	fileBytes, err := os.ReadFile(selinuxConfigFilePath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read existing grub2 config file:\n%w", err)
+	}
+
+	selinuxConfig := string(fileBytes)
+
+	match := selinuxConfigModeRegex.FindStringSubmatch(selinuxConfig)
+	if match == nil {
+		return "", fmt.Errorf("failed to find SELinux mode in %s file", installutils.SELinuxConfigFile)
+	}
+
+	selinuxConfigMode := match[1]
+
+	switch selinuxConfigMode {
+	case "enforcing":
+		return imagecustomizerapi.SELinuxEnforcing, nil
+
+	case "permissive":
+		return imagecustomizerapi.SELinuxPermissive, nil
+
+	case "disabled":
+		return imagecustomizerapi.SELinuxDisabled, nil
+
+	default:
+		return "", fmt.Errorf("unknown SELinux mode (%s) found in (%s) file", selinuxConfigMode,
+			installutils.SELinuxConfigFile)
+	}
+}
+
+func readGrub2ConfigFile(imageChroot *safechroot.Chroot) (string, error) {
+	logger.Log.Debugf("Reading grub.cfg file")
+
+	grub2ConfigFilePath := getGrub2ConfigFilePath(imageChroot)
+
+	// Read the existing grub.cfg file.
+	grub2ConfigFileBytes, err := os.ReadFile(grub2ConfigFilePath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read existing grub2 config file:\n%w", err)
```

Comment:
Might be worth to mention the path we tried to read.

Reply:
fixed